<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="inflation_radius" default="0.2" />
    <arg name="map_resolution" default="0.1" />
    <arg name="localmap_frameid" default="base_link" />

    <arg name="solver_timeout_ms" default="40.0" />
    <arg name="subgoal_timer_interval" default="0.5" />

    <!-- Scan to local map node -->
    <node name="object2localmap_node" pkg="path_finding" type="object2localmap_node" required="true" output="screen">
        <param name="inflation_radius" type="double" value="$(arg inflation_radius)" />
        <param name="map_resolution" type="double" value="$(arg map_resolution)" />
        <param name="localmap_frameid" type="str" value="$(arg localmap_frameid)" />
    </node> 

    <!-- A* path finding node -->
    <node name="path_finding_node" pkg="path_finding" type="path_finding_node" required="true" output="screen">
        <param name="solver_timeout_ms" type="double" value="$(arg solver_timeout_ms)" />
        <param name="subgoal_timer_interval" type="double" value="$(arg subgoal_timer_interval)" />
    </node>

</launch>